File: packages/frontend/src/api/vendor.api.ts
import api from './client';
import type { ApiResponse, OrderItem, VendorCatalogCategory, CategoryAttribute } from '@/types';

export interface VendorResponsePayload {
  action: 'ACCEPT' | 'REJECT' | 'ALTER';
  alteredQty?: number;
  note?: string;
  offeredUnitPrice?: number;
}

export const vendorApi = {
  getIncomingOrders: () =>
    api.get<ApiResponse<OrderItem[]>>('/vendor/orders').then((r) => r.data.data),

  respondToItem: (itemId: string, data: VendorResponsePayload) =>
    api.put<ApiResponse<OrderItem>>(`/vendor/orders/items/${itemId}/respond`, data).then((r) => r.data.data),

  bulkRespond: (
    responses: Array<{
      itemId: string;
      action: 'ACCEPT' | 'REJECT' | 'ALTER';
      alteredQty?: number;
      offeredUnitPrice?: number;
      note?: string;
    }>,
  ) =>
    api
      .post<ApiResponse<{ updated: number }>>('/vendor/orders/items/bulk-respond', { responses })
      .then((r) => r.data.data),

  respondToTraderPrice: (itemId: string, data: { action: 'ACCEPT' | 'REJECT' }) =>
    api.put<ApiResponse<OrderItem>>(`/vendor/orders/items/${itemId}/price-counter`, data).then((r) => r.data.data),

  getCatalogCategories: () =>
    api.get<ApiResponse<VendorCatalogCategory[]>>('/vendor/categories').then((r) => r.data.data),

  createVendorAttribute: (categoryId: string, name: string, sortOrder?: number) =>
    api
      .post<ApiResponse<CategoryAttribute>>(`/vendor/categories/${categoryId}/attributes`, { name, sortOrder })
      .then((r) => r.data.data),

  updateVendorAttribute: (
    categoryId: string,
    attributeId: string,
    data: { name?: string; sortOrder?: number },
  ) =>
    api
      .put<ApiResponse<CategoryAttribute>>(
        `/vendor/categories/${categoryId}/attributes/${attributeId}`,
        data,
      )
      .then((r) => r.data.data),

  deleteVendorAttribute: (categoryId: string, attributeId: string) =>
    api.delete<ApiResponse<null>>(`/vendor/categories/${categoryId}/attributes/${attributeId}`).then((r) => r.data),
};

File: packages/mobile/lib/core/api/vendor_api.dart
import 'package:dio/dio.dart';

import '../network/api_response.dart';

class VendorApi {
  VendorApi(this._dio);
  final Dio _dio;

  Future<List<dynamic>> getIncomingOrders() async {
    final res = await _dio.get<Map<String, dynamic>>('/vendor/orders');
    return ApiResponse.unwrapData(res.data!, (d) => d as List<dynamic>);
  }

  Future<Map<String, dynamic>> respondToItem(String itemId, Map<String, dynamic> data) async {
    final res = await _dio.put<Map<String, dynamic>>('/vendor/orders/items/$itemId/respond', data: data);
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }

  Future<Map<String, dynamic>> bulkRespond(List<Map<String, dynamic>> responses) async {
    final res = await _dio.post<Map<String, dynamic>>(
      '/vendor/orders/items/bulk-respond',
      data: {'responses': responses},
    );
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }

  Future<Map<String, dynamic>> respondToTraderPrice(String itemId, Map<String, dynamic> data) async {
    final res = await _dio.put<Map<String, dynamic>>(
      '/vendor/orders/items/$itemId/price-counter',
      data: data,
    );
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }

  Future<List<dynamic>> getCatalogCategories() async {
    final res = await _dio.get<Map<String, dynamic>>('/vendor/categories');
    return ApiResponse.unwrapData(res.data!, (d) => d as List<dynamic>);
  }

  Future<Map<String, dynamic>> createVendorAttribute(
    String categoryId,
    Map<String, dynamic> body,
  ) async {
    final res = await _dio.post<Map<String, dynamic>>(
      '/vendor/categories/$categoryId/attributes',
      data: body,
    );
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }

  Future<Map<String, dynamic>> updateVendorAttribute(
    String categoryId,
    String attributeId,
    Map<String, dynamic> data,
  ) async {
    final res = await _dio.put<Map<String, dynamic>>(
      '/vendor/categories/$categoryId/attributes/$attributeId',
      data: data,
    );
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }

  Future<void> deleteVendorAttribute(String categoryId, String attributeId) async {
    await _dio.delete<Map<String, dynamic>>('/vendor/categories/$categoryId/attributes/$attributeId');
  }
}
